hushline/settings.py

```diff
@@ -224,57 +224,55 @@ def index() -> str | Response:
         directory_visibility_form = DirectoryVisibilityForm()
         profile_form = ProfileForm()
 
-        # Check if the bio update form was submitted
-        if (
-            request.method == "POST"
-            and "update_bio" in request.form
-            and profile_form.validate_on_submit()
-        ):
-            user.bio = request.form["bio"]
-            user.extra_field_label1 = profile_form.extra_field_label1.data.strip()
-            user.extra_field_value1 = profile_form.extra_field_value1.data.strip()
-            user.extra_field_label2 = profile_form.extra_field_label2.data.strip()
-            user.extra_field_value2 = profile_form.extra_field_value2.data.strip()
-            user.extra_field_label3 = profile_form.extra_field_label3.data.strip()
-            user.extra_field_value3 = profile_form.extra_field_value3.data.strip()
-            user.extra_field_label4 = profile_form.extra_field_label4.data.strip()
-            user.extra_field_value4 = profile_form.extra_field_value4.data.strip()
-            db.session.commit()
-            flash("👍 Bio updated successfully.")
-            return redirect(url_for("settings.index"))
+        # Handle form submissions
+        if request.method == "POST":
+            # Update bio and custom fields
+            if "update_bio" in request.form and profile_form.validate_on_submit():
+                user.bio = request.form["bio"].strip()
+                user.extra_field_label1 = profile_form.extra_field_label1.data.strip()
+                user.extra_field_value1 = profile_form.extra_field_value1.data.strip()
+                user.extra_field_label2 = profile_form.extra_field_label2.data.strip()
+                user.extra_field_value2 = profile_form.extra_field_value2.data.strip()
+                user.extra_field_label3 = profile_form.extra_field_label3.data.strip()
+                user.extra_field_value3 = profile_form.extra_field_value3.data.strip()
+                user.extra_field_label4 = profile_form.extra_field_label4.data.strip()
+                user.extra_field_value4 = profile_form.extra_field_value4.data.strip()
+
+                # Trigger the rel=me verification for each URL field
+                for i in range(1, 5):
+                    url_to_verify = getattr(user, f"extra_field_value{i}", "").strip()
```

Comment:
are these all guaranteed to be valid URLs? Should we do an initial filter to only verify fields that are valid urls using https?

Reply:
Yes I agree. We should check if the field value starts with `https://`, which is the same logic I used in the template to determine if the fields should be links or not:

```jinja2
{% if field.value.startswith('https://') %}
```

Reply:
Changes in 83a47e9

Confirmed in logs:
```
2024-08-30 11:12:05,440:ERROR:Error fetching URL for field 1: glenn.01
2024-08-30 11:12:05,440:ERROR:Error fetching URL for field 2: sciencedesign
2024-08-30 11:12:05,440:ERROR:Error fetching URL for field 3: fosstodon.org/@scidsg
2024-08-30 11:12:05,949:INFO:127.0.0.1 - - [30/Aug/2024 11:12:05] "POST /settings/ HTTP/1.1" 302 -
2024-08-30 11:12:05,959:INFO:127.0.0.1 - - [30/Aug/2024 11:12:05] "GET /settings/ HTTP/1.1" 200 -
2024-08-30 11:12:05,995:INFO:127.0.0.1 - - [30/Aug/2024 11:12:05] "GET /static/css/style.css HTTP/1.1" 304 -
2024-08-30 11:12:05,997:INFO:127.0.0.1 - - [30/Aug/2024 11:12:05] "GET /static/js/global.js HTTP/1.1" 304 -
2024-08-30 11:12:05,998:INFO:127.0.0.1 - - [30/Aug/2024 11:12:05] "GET /static/js/settings.js HTTP/1.1" 304 -
2024-08-30 11:12:06,000:INFO:127.0.0.1 - - [30/Aug/2024 11:12:06] "GET /static/img/app/dropdown.png HTTP/1.1" 304 -
2024-08-30 11:12:06,001:INFO:127.0.0.1 - - [30/Aug/2024 11:12:06] "GET /static/img/app/icon-search.png HTTP/1.1" 304 -
2024-08-30 11:12:06,007:INFO:127.0.0.1 - - [30/Aug/2024 11:12:06] "GET /static/img/app/icon-verified-lm.png HTTP/1.1" 304 -
2024-08-30 11:12:06,032:INFO:127.0.0.1 - - [30/Aug/2024 11:12:06] "GET /static/css/style.css HTTP/1.1" 304 -
2024-08-30 11:15:44,992:INFO: * Detected change in '/Users/glennsorrentino/Library/Mobile Documents/com~apple~CloudDocs/Git/hushline/hushline/settings.py', reloading
2024-08-30 11:15:45,109:INFO: * Restarting with stat
2024-08-30 11:15:45,686:WARNING: * Debugger is active!
2024-08-30 11:15:45,694:INFO: * Debugger PIN: 328-910-462
2024-08-30 11:16:20,111:INFO:127.0.0.1 - - [30/Aug/2024 11:16:20] "GET /settings/ HTTP/1.1" 200 -
2024-08-30 11:16:20,138:INFO:127.0.0.1 - - [30/Aug/2024 11:16:20] "GET /static/js/global.js HTTP/1.1" 304 -
2024-08-30 11:16:20,139:INFO:127.0.0.1 - - [30/Aug/2024 11:16:20] "GET /static/css/style.css HTTP/1.1" 304 -
2024-08-30 11:16:20,139:INFO:127.0.0.1 - - [30/Aug/2024 11:16:20] "GET /static/js/settings.js HTTP/1.1" 304 -
2024-08-30 11:16:20,143:INFO:127.0.0.1 - - [30/Aug/2024 11:16:20] "GET /static/img/app/dropdown.png HTTP/1.1" 304 -
2024-08-30 11:16:20,143:INFO:127.0.0.1 - - [30/Aug/2024 11:16:20] "GET /static/img/app/icon-search.png HTTP/1.1" 304 -
2024-08-30 11:16:20,156:INFO:127.0.0.1 - - [30/Aug/2024 11:16:20] "GET /static/img/app/icon-verified-lm.png HTTP/1.1" 304 -
2024-08-30 11:16:24,087:INFO:127.0.0.1 - - [30/Aug/2024 11:16:24] "POST /settings/ HTTP/1.1" 302 -
2024-08-30 11:16:24,102:INFO:127.0.0.1 - - [30/Aug/2024 11:16:24] "GET /settings/ HTTP/1.1" 200 -
2024-08-30 11:16:24,128:INFO:127.0.0.1 - - [30/Aug/2024 11:16:24] "GET /static/css/style.css HTTP/1.1" 304 -
2024-08-30 11:16:24,143:INFO:127.0.0.1 - - [30/Aug/2024 11:16:24] "GET /static/img/app/dropdown.png HTTP/1.1" 304 -
2024-08-30 11:16:24,145:INFO:127.0.0.1 - - [30/Aug/2024 11:16:24] "GET /static/js/global.js HTTP/1.1" 304 -
2024-08-30 11:16:24,146:INFO:127.0.0.1 - - [30/Aug/2024 11:16:24] "GET /static/js/settings.js HTTP/1.1" 304 -
2024-08-30 11:16:24,148:INFO:127.0.0.1 - - [30/Aug/2024 11:16:24] "GET /static/img/app/icon-verified-lm.png HTTP/1.1" 304 -
2024-08-30 11:16:24,151:INFO:127.0.0.1 - - [30/Aug/2024 11:16:24] "GET /static/img/app/icon-search.png HTTP/1.1" 304 -
```